src/config.c

```diff
@@ -225,21 +228,70 @@ void config_vifs_from_kernel(void)
     struct uvif *v;
     vifi_t vifi;
     uint32_t addr, mask, subnet;
-    struct ifaddrs *ifaddr, *ifa;
+    struct ifaddrs *ifaddr, *ifa,*ifap;
+    int phyint_num;
+    int count;
+    struct iflist *entry;
 
     /* Query config first for list of enabled interfaces */
-    build_iflist();
+    phyint_num = build_iflist();
 
+init_vif_list:
     total_interfaces = 0; /* The total number of physical interfaces */
     if (getifaddrs(&ifaddr) == -1) {
 	logit(LOG_ERR, errno, "Failed retrieving interface addresses");
 	return;
     }
 
+    ifap = calloc(phyint_num, (sizeof(struct ifaddrs)));
+    if(!ifap) {
+	    logit(LOG_ERR,0,"%s[%d]: Allocation error",__func__,__LINE__);
```

Comment:
I'm a bit picky about following the coding style pf surrounding code:

1. `if(...)` --> `if (...)`
2. All arguments should be spaced, like in a regular sentence: `logint(LOG_ERR, 0, "...", __func__, __LINE__);`